core/arch/arm/kernel/kern.ld.S

```diff
@@ -379,7 +374,11 @@ SECTIONS
 
 }
 
-#if defined(CFG_TEE_COHERENT_START)
-ASSERT((__coherent_start & (4096 - 1)) == 0, "Coherent RAM start is not 4Kbyte aligned")
-ASSERT((__coherent_end & (4096 - 1)) == 0, "Coherent RAM end is not 4Kbyte aligned")
+#if CFG_TEE_COHERENT_SIZE
+ASSERT(!(__coherent_start & (4096 - 1)),
+					"Coherent memory start alignment");
+ASSERT(!(CFG_TEE_COHERENT_SIZE & (4096 - 1)),
```

Comment:
The test above explicitly notifies developer that the coherent size follows the mapping constraint. Otherwise, this will be detected only a runtime. I found this explicit trace more convenient, but I agree it is redundant with the runtime assertions/panics.

Test below does not check the alignment but the size. In case one defines CFG_TEE_COHERENT_SIZE to a value below the effective coherent section computed at link stage, the coherent might the only partially mapped and only run time data abort could detect it.